### PR TITLE
Make improvement for supporting other iso with virtualbox

### PR DESCRIPTION
**Pull request by TrumanLing**
Fix issue #669 
Enforce support with other iso, deatail modification:
Dynamically get iso filename
Dynamically set dvddrive medium
Dynamically set docker configration directory

### CODE REVIEW

**Comment by TrumanLing** [comment]
@bfirshv @ehazlett because of PR #693 , do you have plan to merge this PR

**Comment by shawnzhu** [comment]
@TrumanLing @ehazlett do you think it's a good idea to fix option `--virtualbox-boo2docker-url` before adding more feature?

**Comment by TrumanLing** [comment]
@shawnzhu done, thanks for your review. And I agree with you.

**Comment by sthulb** [comment]
All instances of --<driver>-boot2docker-url should just become `--<driver>-image-url` for now.

File: drivers/virtualbox/virtualbox.go [hunk 1]
```diff
@@ -553,9 +551,8 @@ func (d *Driver) diskPath() string {
 // Make a boot2docker VM disk image.
 func (d *Driver) generateDiskImage(size int) error {
 	log.Debugf("Creating %d MB hard disk image...", size)
-
-	magicString := "boot2docker, please format-me"
-
+	parts := strings.Split(isoFilename, ".")
+	magicString := fmt.Sprintf("%s, please format-me", parts[0])
```
Comment (sthulb) [inline drivers/virtualbox/virtualbox.go]: You can't assume that all images will support automatic disk formatting.

Comment (TrumanLing) [inline drivers/virtualbox/virtualbox.go]: @sthulb thanks for your review.I have removed code for dynamically set magic string.

**Comment by sthulb** [comment]
@TrumanLing We're probably going to be stripping out all of the boot2docker stuff soon, in favour of cloudinit.

**Comment by TrumanLing** [comment]
@sthulb I know your idea, but I think, same as what @shawnzhu said, it is better to fix it before addiong new feature.

**Comment by bfirsh** [comment]
Thanks for this @TrumanLing. There aren't any immediate plans to remove boot2docker, so we should certainly still maintain what we have.

**Comment by sthulb** [comment]
My comment was vague I guess, I meant in terms of supporting other distros, we might not have to worry about modifying boot2docker specific code to work with other images.

**Comment by shawnzhu** [comment]
@TrumanLing @sthulb @bfirsh I believe we all want to fix the documented feature `virtualbox-boot2docker-url` because it doesn't work right now. a new PR as fix?

**Comment by ehazlett** [comment]
@shawnzhu it actually does work but with b2d ISOs.  We are actually doing this with the vmwarefusion driver.  They use a custom b2d iso.

@TrumanLing thanks for the contribution but I'm not sure we want to change that much at the core of the virtualbox driver at this time.  Right now, the `--virtualbox-boot2docker-url` supports custom b2d ISOs from version 1.5+.

**Comment by shawnzhu** [comment]
@ehazlett it is the option `--virtualbox-boot2docker-url` that doesn't work. see [Line147 of drivers/virtualbox/virtualbox.go](https://github.com/docker/machine/blob/master/drivers/virtualbox/virtualbox.go#L147), it should uses `imgPath` when saving iso file instead of `commonIsoPath`, or you will get error output like that in #669.

I would prefer to fix the option `--virtualbox-boot2docker-url` under this PR.

**Comment by shawnzhu** [comment]
@ehazlett I created #826 to fix the option `--virtualbox-boot2docker-url` only.

**Comment by ehazlett** [comment]
@shawnzhu thx -- making comments there.

**Comment by nathanleclaire** [comment]
@TrumanLing Please take a look at making a new PR if you would still like to support this feature (this PR is somewhat outdated).

**Pull request closed without merging**